ansible/roles/products/hand-e/server/desktop-icons/tasks/hand-icon.yml

```diff
@@ -5,6 +5,27 @@
     dest: "{{ shadow_hand_launcher_folder }}/{{ desktop_icon_png }}"
     mode: '664'
 
+- name: Set launch file path based on hand_side {{ hand_side }} for getting hand_serial
+  set_fact:
+    launch_file_path: "{{ shadow_hand_launcher_folder }}/sr_{{ hand_side[0] }}hand.launch"
+
+- name: Download the launch file config branch {{ config_branch }} and hand_side {{ hand_side }} from Github
+  get_url:
+    url: "https://raw.githubusercontent.com/shadow-robot/sr-config/{{ config_branch }}/sr_ethercat_hand_config/launch/sr_{{ hand_side[0] }}hand.launch"
+    dest: "{{ launch_file_path }}"
+    mode: '755'
+
+- name: Get the hand_serial xml_response from the launch file
+  xml:
+    path: "{{ launch_file_path }}"
+    xpath: "//attribute::*[contains(., 's{{ hand_side[0] }}hand.launch')]/..//attribute::*[contains(., 'hand_serial')]/.."
```

Comment:
Is this supposed to look like that? Not sr_?

Reply:
yeah, it's supposed to look like that: https://github.com/shadow-robot/sr-config/blob/demohand_A/sr_ethercat_hand_config/launch/sr_rhand.launch#L6

Reply:
but I realised another error: there is never slhand.launch because srhand.launch is "shadow robot hand.launch" not "shadow right hand.launch" https://github.com/shadow-robot/sr_interface/tree/melodic-devel/sr_robot_launch/launch